packages/netsuite-adapter/src/filters/omit_fields.ts

```diff
@@ -14,96 +14,140 @@
+      if (typesForDeepTransformation.has(instance.elemID.typeName)) {
+        instance.value = await transformValues({
+          values: instance.value,
+          type: await instance.getType(),
+          transformFunc,
+          strict: false,
+        }) ?? {}
+      }
+    })
+
+  await awu(elements)
+    .filter(isObjectType)
+    .filter(isCustomRecordType)
+    .forEach(async type => {
+      type.annotations = omitValues(type.annotations, CUSTOM_RECORD_TYPE)
+      if (typesForDeepTransformation.has(CUSTOM_RECORD_TYPE)) {
+        type.annotations = await transformElementAnnotations({
+          element: type,
+          transformFunc,
+          strict: false,
+        })
+      }
+      Object.values(type.fields).forEach(field => {
+        field.annotations = omitValues(field.annotations, CUSTOM_FIELDS_LIST)
+      })
+    })
+}
+
+
+const filterCreator: LocalFilterCreator = ({ config, elementsSource }) => ({
   name: 'omitFieldsFilter',
   onFetch: async elements => {
-    const fieldsToOmit = FIELDS_TO_OMIT.concat(config.fetch?.fieldsToOmit ?? [])
+    const fieldsToOmit = FIELDS_TO_OMIT_ON_FETCH.concat(config.fetch?.fieldsToOmit ?? [])
 
     if (fieldsToOmit.length === 0) {
       return
     }
 
     const typeNames = elements.filter(isObjectType).map(elem => elem.elemID.name).concat(CUSTOM_FIELDS_LIST)
-    const fieldsToOmitByType = Object.fromEntries(
-      typeNames.map((typeName: string): [string, string[]] => [
-        typeName,
-        fieldsToOmit
-          .filter(({ type, subtype }) => isFullRegexMatch(typeName, subtype !== undefined ? subtype : type))
-          .flatMap(params => params.fields),
-      ]).filter(([_t, fields]) => fields.length > 0)
-    )
+    const fieldsToOmitByType = getFieldsToOmitByType(typeNames, fieldsToOmit)
 
     if (_.isEmpty(fieldsToOmitByType)) {
       return
     }
 
-    const typesForDeepTransformation = new Set(
-      typeNames.filter(typeName => fieldsToOmit
-        .some(({ type, subtype }) => subtype !== undefined && subtype !== type && isFullRegexMatch(typeName, type)))
-    )
-
-    const omitValues = (value: Values, typeName: string): Values => (
-      typeName in fieldsToOmitByType
-        ? _.omitBy(value, (_val, key) => fieldsToOmitByType[typeName]
-          .some(fieldToOmit => isFullRegexMatch(key, fieldToOmit)))
-        : value
-    )
-
-    const transformFunc: TransformFunc = async ({ value, field }) => {
-      if (!_.isPlainObject(value)) {
-        return value
-      }
-      const fieldType = await field?.getType()
-      if (!isObjectType(fieldType)) {
-        return value
-      }
-      return omitValues(value, fieldType.elemID.name)
+    const typesForDeepTransformation = getTypesForDeepTransformation(typeNames, fieldsToOmit)
+
+    await omitFieldsFromElements(elements, fieldsToOmitByType, typesForDeepTransformation)
+  },
+  preDeploy: async changes => {
+    const elements = changes.map(getChangeData)
```

Comment:
In modifications we should go over the before & after (otherwise we will get wrong fields to the null list - `data_instances_null_fields.ts`). We can do the following:
```suggestion
    const elements = changes.filter(isAdditionOrModificationChange).flatMap(change => Object.values(change.data))
```